axum-macros/src/debug_handler.rs

```diff
@@ -284,6 +286,169 @@ fn check_inputs_impls_from_request(item_fn: &ItemFn, state_ty: Type) -> TokenStr
         .collect::<TokenStream>()
 }
 
+///If the output is a tuple with 2 or more elements,
+/// it checks with the following pattern,
+/// first element => StatusCode || Parts || IntoResponseParts
+///last element => IntoResponse
+///other elements => IntoResponseParts
+///the max numbers of IntoResponseParts(16)
+fn check_output_tuples(item_fn: &ItemFn) -> Option<TokenStream> {
+    //Extract tuple types
+    let elements = match &item_fn.sig.output {
+        ReturnType::Type(_, ty) => match &**ty {
+            Type::Tuple(tuple) => &tuple.elems,
+            _ => return None,
+        },
+        _ => return None,
+    };
+
+    if elements.len() < 2 {
+        return None;
+    }
+    //Amount of IntoRequestParts
+    let mut parts_amount: i8 = 0;
+
+    let token_stream = WithPosition::new(elements.iter())
+        .enumerate()
+        .map(|(_idx, arg)| match &arg {
+            //First element type in the tuple
+            Position::First(ty) => {
+                let typename = extract_clean_typename(ty);
+                if typename.is_none() {
+                    quote! {}
+                } else {
+                    let typename = typename.unwrap();
+                    match &*typename.to_string() {
+                        "Parts" => quote! {},
+                        "Response" => quote! {},
+                        "StatusCode" => {
+                            quote! {}
+                        }
+                        _ => {
+                            parts_amount += 1;
+                            check_into_response_parts(ty, parts_amount)
+                        }
+                    }
+                }
```

Comment:
I think you can do `let typename = extract_clean_typename(ty).map(...)` instead.